packages/notification-center/src/store/novu-theme-provider.context.tsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { INovuTheme, ThemeContext } from './novu-theme.context';
+import { ColorScheme } from '../index';
+import { defaultDarkTheme, defaultLightTheme } from '../shared/config/themeDefaultValues';
+
+export interface INovuProvider {
+  colorScheme: ColorScheme;
+  light?: INovuTheme;
+  dark?: INovuTheme;
+  common?: ICommonTheme;
+}
+
+interface ICommonTheme {
+  fontFamily: string;
+}
+
+interface INovuThemeProviderProps {
+  children: React.ReactNode;
+  theme: INovuProvider;
+}
+
+export function NovuThemeProvider(props: INovuThemeProviderProps) {
+  const novuTheme =
+    props.theme.colorScheme === 'light'
+      ? { ...defaultLightTheme, ...props.theme.light }
+      : { ...defaultDarkTheme, ...props.theme.dark };
```

Comment:
Imagine the following use case where I want to override this darktheme:
```
<PopoverComponent theme={{ layoutWrap: { colors: { main: 'white' }}}}></...>
```

I would suggest doing a deepMerge here to make sure only partial updates will be used. 

this will effectively delete all the other props there: "fontColor", "secondaryFontColor" or will force the user to provide all of the data there.

Reply:
Totally agree I missed it on the conversion to objects